### PR TITLE
Reverted all instances of removed gas estimations

### PR DESCRIPTION
**Pull request by ug02fast**
Brought back gas estimations
image

https://github.com/saddle-finance/saddle-frontend/issues/760

### CODE REVIEW

File: src/utils/gas.ts [hunk 1]
```diff
@@ -15,12 +15,12 @@ export function gasBNFromState(
   let gasPrice
   if (gasSelected === GasPrices.Custom) {
     gasPrice = gasCustom?.valueSafe
-  } else if (gasSelected === GasPrices.Fast) {
-    gasPrice = gasFast
+  } else if (gasSelected === GasPrices.Standard) {
+    gasPrice = gasStandard
```
Comment (ug02fast) [inline src/utils/gas.ts:19]: Does the default to fast have to be adjusted here @hammeiam ?

File: src/state/user.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const initialState: UserState = {
   userSwapAdvancedMode: false,
   userPoolAdvancedMode: false,
   userDarkMode: false,
-  gasPriceSelected: GasPrices.Standard,
+  gasPriceSelected: GasPrices.Fast,
```
Comment (ug02fast) [inline src/state/user.ts:48]: These are a couple places I found throughout the codebase where defaults are set. Are there any others? @hammeiam

File: src/hooks/useApproveAndSwap.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export function useApproveAndSwap(): (
       } else {
         gasPrice = gasStandard
       }
-      gasPrice = parseUnits(gasPrice ? String(gasPrice) : "45", 9)
+      gasPrice = parseUnits(String(gasPrice), 9)
```
[the remark this answers is not in the document]
Comment (ug02fast) [reply src/hooks/useApproveAndSwap.ts:92]: Yeah this number represents the normal gas price. We are tying to move away from hardcoding.

[the remark this answers is not in the document]
Comment (ug02fast) [reply src/hooks/useApproveAndSwap.ts:92]: @nkrulikovsky Sorry I meant to say we are moving away from defaulting to standard Gas and towards fastGas. 45 used to represent standard gas but after EIP-1559 that isn't necessary.

File: src/components/ReviewDeposit.tsx [hunk 4]
```diff
@@ -123,7 +123,7 @@ function ReviewDeposit({
             </span>
           </div>
         )}
-        {transactionData.txnGasCost?.valueUSD && (
+        {transactionData.txnGasCost?.valueUSD && shouldDisplayGas && (
```
Comment (hammeiam) [inline src/components/ReviewDeposit.tsx:126]: is `valueUSD` a BigNumber? If so, `shouldDisplayGas` is probably redundant and we can just do `valueUSD.eq(Zero)`

Comment (ug02fast) [reply src/components/ReviewDeposit.tsx]: good point. i did something similar using `gt` to avoid changing too much TS.

**Comment by hammeiam** [comment]
Following a discussion, we're deciding to close this given that we don't yet have a good story around cross-chain gas estimation, and would prefer to push that responsibility to the user's wallet. We may come back to this once that's no longer the case.

**Pull request closed without merging**